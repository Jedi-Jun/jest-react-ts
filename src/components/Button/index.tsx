import styled from 'styled-components';

interface ContainerProps {
  readonly backgroundColor?: string;
  readonly hoverColor?: string;
}

const ButtonStyle = styled.button<ContainerProps>`
  background: ${(props) => props.backgroundColor || '#304ffe'};
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  border: 0;
  &: hover {
    background: ${(props) => props.hoverColor || '#1e40ff'};
  }
  &: active {
    box-shadow: inset 10px 10px 20px rgba(0, 0, 0, 0.2);
  }
`;
const Label = styled.div`
  color: #ffffff;
  font-size: 16px;
`;

interface Props {
  readonly label: string;
  readonly backgroundColor?: string;
  readonly hoverColor?: string;
  readonly onClick?: () => void;
}

export const Button = ({ onClick, label, backgroundColor, hoverColor }: Props) => {
  return (
    <ButtonStyle onClick={onClick} backgroundColor={backgroundColor} hoverColor={hoverColor}>
      <Label>{label}</Label>
    </ButtonStyle>
  );
};
