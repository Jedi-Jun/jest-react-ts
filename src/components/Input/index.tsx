import React from 'react';
import styled from 'styled-components';

const InputStyle = styled.input.attrs({
  type: 'text',
})`
  display: flex;
  font-size: 16px;
  padding: 8px 10px;
  border-radius: 8px;
  border: 3px solid #bdbdbd;
  outline: none;
`;

interface Props {
  readonly value?: string;
  readonly onChange?: (text: string) => void;
  readonly placeholder?: string;
  readonly onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
}

export const Input = ({ value, onChange, placeholder, onKeyPress }: Props) => {
  const handlerChange = (event) => {
    if (typeof onChange === 'function') {
      onChange(event.target.value);
    }
  };
  return (
    <InputStyle
      value={value}
      placeholder={placeholder}
      onChange={handlerChange}
      onKeyPress={onKeyPress}
    />
  );
};
