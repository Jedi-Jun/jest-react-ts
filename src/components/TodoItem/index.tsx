import styled from 'styled-components';
import { Button } from 'components';

const Container = styled.div`
  display: flex;
  border: 1px solid blue;
  align-items: center;
  justify-contents: center;
  // align-items: bottom;
  // padding: 10px;
`;

const Label = styled.label`
  flex: 1;
  font-size: 1.3rem;
  margin-right: 10px;
  border-top: 1px solid blue;
`;

interface Props {
  readonly label: string;
  readonly onDelete?: () => void;
}

export const ToDoItem = ({ label, onDelete }: Props) => {
  return (
    <Container>
      <Label>{label}</Label>
      <Button label="Del" backgroundColor="#ff1744" hoverColor="#d01440" onClick={onDelete} />
    </Container>
  );
};
