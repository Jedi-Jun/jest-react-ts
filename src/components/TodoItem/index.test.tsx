import { render, screen, fireEvent } from '@testing-library/react';
import { ToDoItem } from 'components';
import 'jest-styled-components';

describe('<ToDoItem />', () => {
  // #1
  it('renders works correctly', () => {
    const { container } = render(<ToDoItem label="default value" />);
    const todoItem = screen.getByText('default value');
    expect(todoItem).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  // #2
  it('clicks the delete button', () => {
    const handleClick = jest.fn();
    render(<ToDoItem label="default value" onDelete={handleClick} />);
    const deleteButton = screen.getByText('Del');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(deleteButton);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
