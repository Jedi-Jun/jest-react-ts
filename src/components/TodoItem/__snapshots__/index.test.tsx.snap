// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoItem /> renders works correctly 1`] = `
.c2 {
  background: #ff1744;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  border: 0;
}

.c2:hover {
  background: #d01440;
}

.c2:active {
  box-shadow: inset 10px 10px 20px rgba(0,0,0,0.2);
}

.c3 {
  color: #ffffff;
  font-size: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid blue;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-contents: center;
  -ms-flex-pack: center;
  justify-contents: center;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.3rem;
  margin-right: 10px;
  border-top: 1px solid blue;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      default value
    </label>
    <button
      class="c2"
    >
      <div
        class="c3"
      >
        Del
      </div>
    </button>
  </div>
</div>
`;
